File: app/src/main/java/io/jasonatwood/networkmonkey/NoOpNetworkMonkey.java
package io.jasonatwood.networkmonkey;

import java.io.IOException;

import okhttp3.Request;
import okhttp3.Response;

/**
 * NetworkMonkey that does nothing. Just passes request straight through.
 */
public class NoOpNetworkMonkey implements NetworkMonkey {

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        return chain.proceed(request);
    }

    @Override
    public void shouldMonkeyWithWifiConnection() {
        // noop
    }

    @Override
    public void shouldMonkeyWithResponseCode() {
        // noop
    }

    @Override
    public void shouldMonkeyWithResponseTime(int delayInMilliseconds) {
        // noop
    }

    @Override
    public void shouldMonkeyWithRequestSuccess() {
        // noop
    }

    @Override
    public void enableJerkMode() {
        // noop
    }
}

File: networkmonkey/src/main/java/io/jasonatwood/networkmonkey/NoOpNetworkMonkey.kt
package io.jasonatwood.networkmonkey

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * NetworkMonkey that does nothing. Just passes request straight through.
 */
class NoOpNetworkMonkey : NetworkMonkey {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        return chain.proceed(request)
    }

    override fun shouldMonkeyWithWifiConnection() {
        // noop
    }

    override fun shouldMonkeyWithResponseCode(code: Int) {
        // noop
    }

    override fun shouldMonkeyWithResponseTime(delayInMilliseconds: Int) {
        // noop
    }

    override fun shouldMonkeyWithRequestSuccess() {
        // noop
    }

    override fun enableJerkMode() {
        // noop
    }
}